networkit/cpp/graph/GraphTools.cpp

```diff
@@ -200,15 +200,66 @@ double density(const Graph &G) noexcept {
     return m / (n * (n - 1));
 }
 
-double volume(const Graph &G) {
-    if (G.isDirected())
-        throw std::runtime_error("Volume computation is only supported for undirected graphs.");
+double inVolume(const Graph &G, const std::unordered_set<node> &nodes) {
+    double inVolume = 0.0;
+    bool directed = G.isDirected() ? false : true;
```

Comment:
Do you mean `bool directed = !G.isDirected()`? If yes, this is extremely confusing. Please rename the variable to `undirected`.